### ISSUE TITLE
[type-annotation-spacing] Doesn't catch type spacing on optional arguments

### ISSUE THREAD

**Issue by OliverNChalk** [typescript-eslint/typescript-eslint#2955]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
module.exports = {
    root: true,
    parser: "@typescript-eslint/parser",
    parserOptions: {
        project: "tsconfig.json",
        sourceType: "module"
    },
    plugins: ["@typescript-eslint"],
    rules: { "@typescript-eslint/type-annotation-spacing": "error" },
};

```

```TS
class Test
{
    private mTest1: number | undefined;

    public constructor(aTest1? : number)  // No complaints
    {
        this.mTest1 = aTest1;
    }

    private TestMethod(aTest2 : number, aTest1? : number): void  // aTest2 complains, aTest1 does not
    {
        Math.exp(aTest1!);
        return;
    }
}
```
TSConfig:
```
{
    "compilerOptions": {
        "target": "ESNEXT",
        "module": "commonjs",
        "strict": true,
        "noUnusedLocals": false,
        "outDir": "./Dist",
        "removeComments": false,
        "preserveConstEnums": true,
        "inlineSourceMap": true,
        "declaration": true,
        "baseUrl": ".",
        "moduleResolution": "node",
        "esModuleInterop": true
    },    
}

```

**Expected Result**
I expected `type-annotation-spacing` to complain about the space in front of the colon following my optional arguments.

**Actual Result**
It only complains on mandatory arguments.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.18.0` |
| `node`                             | `15.3.0` |

**Comment by bradzacher** [comment]
The lint rule only enforces spacing before the `?` and after the `:` (for optional annotations).

It does not enforce any spacing between the two characters.

**Comment by OliverNChalk** [comment]
The rule says it disallows spaces before the `:` but here I have clearly

**Comment by OliverNChalk** [comment]
This rule says it disallows spaces before the `:` but here I clearly have a space before the `:` on an optional argument. Is that really the intended functionality, for the spacing to be allowed simply because `aArgument` changed to `aArgument?`?

If so, is there any way to achieve my desired functionality? I couldn't find rules related to banning spaces after `?`, which makes sense to me.

**Comment by bradzacher** [comment]
You're misunderstanding the meaning of the code.
The `?` is not owned by the identifier - it's owned by the property definition - the colon.
I.e. the code did not change from `aArgument` to `aArgument?`, it changed from `:` to `?:`.

---

We don't have any rules that enforce what you want.

We avoid spending too much time on stylistic rules as there's no way to keep everyone happy, and they are really complex to build and test.

Our advice is to use a tool made for formatting code - like [prettier](https://github.com/prettier/prettier).
